scipy/optimize/minpack.py

```diff
@@ -720,7 +721,33 @@ def check_gradient(fcn, Dfcn, x0, args=(), col_deriv=0):
     return (good, err)
 
 
-def fixed_point(func, x0, args=(), xtol=1e-8, maxiter=500):
+def _del2(p0, p1, d):
+    return p0 - np.square(p1 - p0) / d
+
+
+def _relerr(actual, desired):
+    return (actual - desired) / desired
+
+
+def _fixed_point_helper(func, x0, args, xtol, maxiter, use_accel):
+    p0 = x0
+    for iter in range(maxiter):
+        p1 = func(p0, *args)
+        if use_accel:
+            p2 = func(p1, *args)
+            d = p2 - 2.0 * p1 + p0
+            p = _lazywhere(d != 0, (p0, p1, d), f=_del2, fillvalue=p2)
+        else:
+            p = p1
+        relerr = _lazywhere(p0 != 0, (p, p0), f=_relerr, fillvalue=p)
+        if np.all(np.abs(relerr) < xtol):
+            return p
+        p0 = p
+    msg = "Failed to converge after %d iterations, value is %s" % (maxiter, p)
+    raise RuntimeError(msg)
```

Comment:
The fixed-point solver uses an iteration more like f(f(...f(f(x0))...)) whereas your linked expectation code I think is more like sum_i(f(i)*p(i)) which seems to be more 'blockable'. The only way I could imagine using a block-like approach would be to check for convergence less frequently, so the fixed-point iteration could run blindly for blocksize steps at a time before checking convergence. I'm not sure this would be a good idea, but maybe that is what you mean?